dmutils/content_loader.py

```diff
@@ -496,16 +607,16 @@ def _load_question(question, directory):
     question_content = read_yaml(
         os.path.join(directory, '{}.yml'.format(question))
     )
-    if question_content:
-        question_content["id"] = _make_question_id(question)
+
+    question_content["id"] = _make_question_id(question)
 
     return question_content
 
 
-def _make_section_id(name):
+def _make_slug(name):
     return inflection.underscore(
         re.sub(r"\s", "_", name)
-    )
+    ).replace('_', '-')
```

Comment:
This will have effect previously existing URLs. I don't think this will have any actual user impact because they should all be dead URLs anyway but worth having a think about.